exp-player/addon/randomizers/pref-phys.js

```diff
@@ -10,105 +12,1336 @@ function shuffleArray(array) {
+        ['cup', 'whiteball', 'lotion', 'spray', 'apple', 'orangeball'],
+        ['orangeball', 'cup', 'lotion', 'spray', 'apple', 'whiteball'],
+        ['apple', 'whiteball', 'lotion', 'cup', 'spray', 'orangeball'],
+        ['whiteball', 'apple', 'orangeball', 'spray', 'cup', 'lotion'],
+        ['cup', 'whiteball', 'orangeball', 'spray', 'apple', 'lotion'],
+        ['orangeball', 'lotion', 'apple', 'whiteball', 'cup', 'spray'],
+        ['whiteball', 'orangeball', 'lotion', 'spray', 'apple', 'cup'],
+        ['whiteball', 'cup', 'lotion', 'spray', 'apple', 'orangeball'],
+        ['apple', 'cup', 'orangeball', 'whiteball', 'lotion', 'spray'],
+        ['orangeball', 'lotion', 'apple', 'cup', 'spray', 'whiteball'],
+        ['apple', 'lotion', 'cup', 'spray', 'whiteball', 'orangeball'],
+        ['cup', 'orangeball', 'lotion', 'spray', 'apple', 'whiteball'],
+        ['whiteball', 'orangeball', 'cup', 'apple', 'spray', 'lotion'],
+        ['cup', 'lotion', 'whiteball', 'spray', 'apple', 'orangeball'],
+        ['cup', 'orangeball', 'lotion', 'spray', 'whiteball', 'apple'],
+        ['lotion', 'cup', 'orangeball', 'spray', 'whiteball', 'apple'],
+        ['lotion', 'orangeball', 'cup', 'whiteball', 'apple', 'spray'],
+        ['apple', 'whiteball', 'cup', 'lotion', 'spray', 'orangeball'],
+        ['cup', 'lotion', 'apple', 'spray', 'whiteball', 'orangeball'],
+        ['apple', 'lotion', 'orangeball', 'spray', 'whiteball', 'cup'],
+        ['whiteball', 'cup', 'orangeball', 'apple', 'spray', 'lotion'],
+        ['whiteball', 'apple', 'orangeball', 'spray', 'lotion', 'cup'],
+        ['whiteball', 'cup', 'lotion', 'apple', 'spray', 'orangeball'],
+        ['apple', 'orangeball', 'cup', 'spray', 'whiteball', 'lotion'],
+        ['whiteball', 'apple', 'orangeball', 'cup', 'lotion', 'spray'],
+        ['apple', 'whiteball', 'orangeball', 'spray', 'cup', 'lotion'],
+        ['cup', 'apple', 'orangeball', 'lotion', 'whiteball', 'spray'],
+        ['cup', 'orangeball', 'apple', 'lotion', 'whiteball', 'spray'],
+        ['whiteball', 'cup', 'orangeball', 'lotion', 'spray', 'apple'],
+        ['orangeball', 'whiteball', 'apple', 'cup', 'spray', 'lotion'],
+        ['whiteball', 'orangeball', 'apple', 'lotion', 'spray', 'cup'],
+        ['cup', 'orangeball', 'whiteball', 'apple', 'spray', 'lotion'],
+        ['apple', 'whiteball', 'lotion', 'spray', 'cup', 'orangeball'],
+        ['orangeball', 'cup', 'whiteball', 'spray', 'apple', 'lotion'],
+        ['orangeball', 'apple', 'cup', 'lotion', 'whiteball', 'spray'],
+        ['cup', 'apple', 'lotion', 'spray', 'whiteball', 'orangeball'],
+        ['orangeball', 'apple', 'whiteball', 'spray', 'cup', 'lotion'],
+        ['cup', 'orangeball', 'whiteball', 'lotion', 'spray', 'apple'],
+        ['orangeball', 'whiteball', 'apple', 'cup', 'lotion', 'spray'],
+        ['whiteball', 'lotion', 'orangeball', 'apple', 'spray', 'cup'],
+        ['whiteball', 'lotion', 'orangeball', 'apple', 'cup', 'spray']
```

Comment:
How was the original list generated? I assume not by hand? Can we use the code you used for the best and most direct comparison?

Insofar as some of these lists were manually filtered (not 100% algorithmically determined), that does add complexity.

I'm tentatively willing to merge this now insofar as we plan to break lookit-specific code out of the shared codebase. (people running other experiments shouldn't need to download this list). The best path forward would be:

1. If you already have the function used to generate the list, then the lists of permutations that *can* be created using that function should be. (eg `conceptOrderRotation`). Most of the permutation-generating algorithms are deterministic and would generate the same order of arrangements every time.

2. We'll leave the "manually curated" lists of permutations as is, because this requires knowledge beyond a brute force algorithm.


You make a good point about clarity and reproducibility, but the tradeoff in those long hard-coded lists  is maintainability; if we use a list of 720 lines * 6 strings each, there's some concern that a single stray typo could change the results and be very hard to track down. 

I think a reliable procedure might be less vulnerable to errors than pure hard-coding. Would that be a viable solution?

Pure technical aside: If we are worried about the performance penalty of generating the permutations, there do seem to be ways to predictably get the n-th permutation. I just wouldn't go tacking them in untested an hour before a holiday release, is all. ;)
http://stackoverflow.com/a/7919887/1422268